### DISCUSSION TITLE
Copilot suggesting not existing methods

### DISCUSSION THREAD

**Discussion by Globolo001** [community/community#32010] [Copilot Conversations]
I use Copilot in Java and I often encounter that it suggests contextually correct code, which is syntactically incorrect.

Example suggestion:
`myObject.contains(...); // 'contains()' doesn't exist`
Correct use from IntelliSense as written in the respective class:
`myObject.containsFile(...); // The correct method`

It seems to me that Copilot doesn't index or use the IntelliSense suggestions. The additional usage of IntelliSense with the AI would strongly improve my experience with this awesome tool.

Or am I missing something?

**Comment by BLKKKBVSIK** [comment]
CoPilot is only suggesting method names and properties depending on the data of your project and the data it collected from user repositories.
Since it's only suggesting, it's quite common that he'll suggest your methods that don't exist since they might not exist in your project/use context, but has been used on other user's repositories

**Comment by Matt-17** [comment]
I don't know about the Java part, but in C# this usually works pretty good. As I understand, Copilot tries to get and understand those methods you have but also needs to suggest new methods if there is no useful method provided for the current task. Of course it would be useful to have a setting to only use existing methods or be creative.

**Comment by Globolo001** [comment]
Crazy how much AI has improved in those few years. 

Agentic (assisted) coding really changed the game.

**Reply by endasil** [reply]
Yet 4 years later it still suggests methods that does not exist. I changed a GetById to GetByIdAsync, now it suggests changing the name of every damn function call i type in in that method to add async at the end, even though no such method exist.
